tests/platform_tests/link_flap/test_cont_link_flap.py

```diff
@@ -124,10 +139,34 @@ def test_cont_link_flap(self, request, duthosts, nbrhosts, enum_rand_one_per_hws
 
             pytest.fail(str(failmsg))
 
+        # Wait 30s for the memory usage to be stable
+        time.sleep(30)
+
         # Record memory status at end
         memory_output = duthost.shell("show system-memory")["stdout"]
         logging.info("Memory Status at end: %s", memory_output)
 
+        # Check the FRR daemons memory usage at end
+        end_time_frr_daemon_memory = {}
+        incr_frr_daemon_memory_threshold = 10 if tbinfo["topo"]["type"] in ["m0", "mx"] else 5
+        for daemon in frr_demons_to_check:
+            # Record FRR daemon memory status at end
+            end_time_frr_daemon_memory[daemon] = self.get_frr_daemon_memory_usage(duthost, daemon)
+            logging.info(f"{daemon} memory usage at end: \n%s", end_time_frr_daemon_memory[daemon])
+
+            # Calculate diff in FRR daemon memory
+            incr_frr_daemon_memory = \
+                float(end_time_frr_daemon_memory[daemon]) - float(start_time_frr_daemon_memory[daemon])
+            logging.info(f"{daemon} absolute difference: %d", incr_frr_daemon_memory)
+
+            # Check FRR daemon memory only if it is increased else default to pass
+            if incr_frr_daemon_memory > 0:
+                percent_incr_frr_daemon_memory = \
+                    (incr_frr_daemon_memory / float(start_time_frr_daemon_memory[daemon])) * 100
+                logging.info(f"{daemon} memory percentage increase: %d", percent_incr_frr_daemon_memory)
```

Comment:
May you please share the typical value for this, thanks

Reply:
> May you please share the typical value for this, thanks

This is the memory usage of a 4700 platform with t1-56-lag topology:
before link flap:
```
INFO     root:test_cont_link_flap.py:34 zebra memory status: 
System allocator statistics:
  Total heap allocated:  19 MiB
  Holding block headers: 225 MiB
  Used small blocks:     0 bytes
  Used ordinary blocks:  12 MiB
  Free small blocks:     2320 bytes
  Free ordinary blocks:  7222 KiB
  Ordinary blocks:       15757
  Small blocks:          69
  Holding blocks:        29

INFO     root:test_cont_link_flap.py:34 bgpd memory status: 
System allocator statistics:
  Total heap allocated:  59 MiB
  Holding block headers: 165 MiB
  Used small blocks:     0 bytes
  Used ordinary blocks:  56 MiB
  Free small blocks:     2608 bytes
  Free ordinary blocks:  2969 KiB
  Ordinary blocks:       5828
  Small blocks:          61
  Holding blocks:        115
```
after link flap:
```
INFO     root:test_cont_link_flap.py:34 zebra memory status: 
System allocator statistics:
  Total heap allocated:  22 MiB
  Holding block headers: 225 MiB
  Used small blocks:     0 bytes
  Used ordinary blocks:  12 MiB
  Free small blocks:     2640 bytes
  Free ordinary blocks:  10297 KiB
  Ordinary blocks:       15823
  Small blocks:          72
  Holding blocks:        29

INFO     root:test_cont_link_flap.py:34 bgpd memory status: 
System allocator statistics:
  Total heap allocated:  60 MiB
  Holding block headers: 164 MiB
  Used small blocks:     0 bytes
  Used ordinary blocks:  57 MiB
  Free small blocks:     2000 bytes
  Free ordinary blocks:  3018 KiB
  Ordinary blocks:       2740
  Small blocks:          56
  Holding blocks:        114
```